ga4gh/backend.py

```diff
@@ -209,6 +214,83 @@ def getReadGroupSets(self):
         """
         return list(self._readGroupSetIdMap.values())
 
+    def getReferenceSets(self):
+        """
+        Returns the list of ReferenceSets in this backend
+        """
+        return list(self._referenceSetIdMap.values())
+
+    def getReference(self, id_):
+        """
+        Returns a reference with the given id_
+        """
+        return self.runGetRequest(self._referenceIdMap, id_)
+
+    def getReferenceSet(self, id_):
+        """
+        Returns a referenceSet with the given id_
+        """
+        return self.runGetRequest(self._referenceSetIdMap, id_)
+
+    def listReferenceBases(self, id_, requestArgs):
+        # parse arguments
+        try:
+            reference = self._referenceIdMap[id_]
+        except KeyError:
+            raise exceptions.ObjectWithIdNotFoundException(id_)
+        start = 0
+        end = datamodel.PysamDatamodelMixin.fastaMax
```

Comment:
Ah right; we can't call `validate` on the dict as it's just a string:string map, and there's no type conversion done. OK, if there are no other GET requests with multiple typed arguments (are there, or will there be?) then I guess what you have is as good as we need. If there are, then I think it's worthwhile thinking about how to generalise this code, and getting automated parsing and type checking of these values from the schema. We can open an issue on this if you would prefer not to slow this PR down any more.